src/components/Dropdown/asyncDropdown/AsyncDropdown.tsx

```diff
@@ -0,0 +1,241 @@
+  };
+
+  const handleChange = (selectedItem: ApiType) => {
+    setSelectedItem(selectedItem);
+    setInputValue(labelField ? itemToString(selectedItem, labelField) : selectedItem.title);
+    onChange(selectedItem);
+
+    if (children || clearInputField) {
+      setInputValue('');
+    }
+  };
+
+  const handleStateChange = (changes: StateChangeOptions<ApiType>) => {
+    const { type } = changes;
+    if (type === Downshift.stateChangeTypes.keyDownEnter) {
+      setInputValue('');
+    }
+  };
+
+  const inputProps = {
+    placeholder,
+    onChange: handleInputChange,
+    onClick,
+    value: inputValue,
+    onBlur,
+  };
+
+  const getOnKeydown = (createOnEnter?: boolean) => {
+    return (event: KeyboardEvent) => {
+      if (event.key === 'Enter') {
+        event.preventDefault();
+        createOnEnter && handleCreate();
+      }
+      if (event.key === 'ArrowDown') {
+        setKeepOpen(true);
+      }
+    };
+  };
+  const handleCreate = () => {
+    if (onCreate) {
+      onCreate(inputValue);
+    }
+    if (children || clearInputField) {
+      setInputValue('');
+    }
+  };
+
+  return (
+    <Downshift
+      itemToString={item => itemToString(item, labelField)}
+      onStateChange={handleStateChange}
+      onChange={handleChange}
+      initialIsOpen={startOpen}
+      selectedItem={selectedItem}
+      defaultIsOpen={keepOpen}
+      onOuterClick={() => {
+        setKeepOpen(false);
+      }}>
+      {({ getInputProps, openMenu, ...downshiftProps }) => {
+        const inpProps = getInputProps({
+          ...inputProps,
+          onKeyDown: getOnKeydown(saveOnEnter && downshiftProps.highlightedIndex === null),
+        });
+
+        return (
+          <div style={positionAbsolute ? { position: 'relative' } : undefined}>
+            {children ? (
+              children({ selectedItems, removeItem, ...inpProps })
+            ) : (
+              <Input
+                {...inpProps}
+                data-testid={'dropdownInput'}
+                iconRight={loading ? <Spinner size="normal" margin="0" /> : <Search />}
+              />
+            )}
+            <DropdownMenu
+              idField={idField}
+              labelField={labelField}
+              multiSelect={multiSelect}
+              selectedItems={selectedItems}
+              disableSelected={disableSelected}
+              {...downshiftProps}
+              items={items}
+              totalCount={totalCount}
+              positionAbsolute={positionAbsolute}
+              onCreate={onCreate && handleCreate}
+              customCreateButtonText={customCreateButtonText}
+              hideTotalSearchCount={hideTotalSearchCount}
+              page={showPagination && page}
+              handlePageChange={handlePageChange}
+            />
+          </div>
+        );
+      }}
+    </Downshift>
+  );
+};
+export default injectT(AsyncDropdown) as <SearchResult, ApiType>(
+  props: Props<SearchResult, ApiType>,
+) => any;
```

Comment:
Denne konverteringen blir ikke helt riktig (hovre over AsyncDropdown i vscode for å se den infererte typen). Det ser ut som type konverteringen bare er nødvendig pga injectT, men t brukes ikke i komponenten så det er bare å fjerne injectT og as. Ser da at det dukker opp en type feil i ContentField og kanskje andre steder, så det bør fikses